src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -58,6 +58,11 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        private bool IsPinOpen(int pin)
```

Comment:
If this is only used on unregistering callbacks, the controller is already ensuring that the pin is open so this would be doing it twice.

Reply:
No. Note that (despite it's name) the _sysFSModes dictionary contains the modes of _all_ pins, not only the ones in use by SysFs. I'll fix that name.

Reply:
@joperezr Ok, now I'm getting what you mean. It's a bit inconsistent tough that the controller and the drivers use different semantics here. All existing driver implementations explicitly allow duplicate opening/closing of pins, the controller however does not. What should be the correct behavior? 
Removing all (or most) validations from the driver is probably also not so nice, since these classes are public, too and could be used by themselves.

Reply:
The idea is that the controller will do all the validation that is in it's power to do, in order to avoid code duplication. This includes checking weather a pin is open or not when trying to peform operations with it, since the controller has a list of all the pins that are opened, so it is able to know if the request is valid or not. The drivers in the other hand are responsible of performing all the validation that is driver-specific, for example, if you try to open pin 100 on a Raspberry pi driver, then the controller won't  know if this is a valid pin number, but the raspberry pi driver will know that this is invalid, so it should throw. That is at least the model we have tried to follow so far, so that is why you still see some validation being made on the driver side, in theory that should only happen when it is a driver-specific check.